File: pautas-api/src/main/java/com/nunesd66/pautas_api/voto/service/VotoService.java
package com.nunesd66.pautas_api.voto.service;

import com.nunesd66.pautas_api.voto.exception.VotoException;
import com.nunesd66.pautas_api.voto.model.Voto;

public interface VotoService {
    void votar(Voto voto) throws VotoException;
}

File: pautas-ui/src/app/voto/service/voto.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Voto } from "../model/voto";
import { Observable } from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class VotoService {

    back_api_url = 'http://localhost:3841/voto';

    constructor(private http: HttpClient) {}

    public votar(voto: Voto): Observable<void> {
        return this.http.post<void>(`${this.back_api_url}/votar`, voto);
    }

}